pkg/config/config.go

```diff
@@ -404,50 +400,15 @@ func (rsc *RedSkyConfig) tokenSource(ctx context.Context) (oauth2.TokenSource, e
 	return nil, nil
 }
 
-// PublicKey attempts to fetch a public key with the specified identifier from the JWKS of the current server
-func (rsc *RedSkyConfig) PublicKey(ctx context.Context, keyID interface{}) (interface{}, error) {
-	// Make sure the key identifier is a string
-	kid, ok := keyID.(string)
-	if !ok {
-		return nil, fmt.Errorf("expected string key identifier")
-	}
-
+// PublicKey attempts to fetch the JWKS of the current server
+func (rsc *RedSkyConfig) PublicKeys(ctx context.Context) (interface{}, error) {
```

Comment:
Couple of things:
1. This function should probably be renamed `FetchJWKS` and just return `(jwk.*Set, error)`
2. The context isn't used any more: aside from allowing the fetch operation to be cancellable, the original intent was to ensure the transport from the OAuth2 client was leveraged (the thought being the same client used for fetching tokens should be used for fetching keys). At the very least it should probably do: `jwk.Fetch(srv.Authorization.JSONWebKeySetURI, jwk.WithHTTPClient(oauth2.NewClient(ctx, nil))`.
3. I'm questioning if we even need this function: it basically be eliminated down to `CurrentServer(...)...` / `jwk.Fetch(...)` and it's only used in one place, maybe we just get rid of it altogether.

Reply:
There's a little overlap with check config and login, so I'm curious if it's worth changing this function to something like
```
func Claims(ctx, oauth2.Token) (map[string]interface, error) {
   set, err := jwk.Fetch(srv.Authorization.JSONWebKeySetURI)
   tokenBytes, err := jws.VerifyWithJWKSet([]byte(t.AccessToken),set, nil)
 	if err != nil {
 		return err
 	}

 	token, err := jwt.ParseBytes(
 		tokenBytes,
 		jwt.WithValidate(true),
 	)
 	if err != nil {
 		return err
 	}

 	return token.PrivateClaims()
}
```

Reply:
Oh, just saw the comment in https://github.com/thestormforge/optimize-controller/pull/261. 

We could add in jwt.options as a variadic parameter to the function signature to account for the validation.

Reply:
I just commented on check config: that should be unvalidated access to the token claims, there is no need to fetch the web keys and validate the token there. In fact: the token really should be opaque (i.e. the check config shouldn't be parsing the token at all: it should be requesting information from a "user info" endpoint).

As for the verification done by login: to the same extent, that validation is really just to prevent presenting the wrong token later on in the process (i.e. fail the login early if validation fails so we don't record the wrong token). Same goes for the "default" namespace check: the token is fine, but if the custom claim isn't set we are just going to get annoying failures later and it requires a either a logout/login or forced login by the user to remedy (i.e. poor experience, lots of support questions).